src/main/groovy/actions/TransferActions.groovy

```diff
@@ -102,4 +105,19 @@ class TransferActions {
     private static updateLabware(destinationLabware) {
         LabwareActions.updateLabware(destinationLabware)
     }
+
+    private static validateLocations(locations, destinationLabware) {
+        def occupiedLocations = locations.collect { location ->
+            def receptacle = destinationLabware.receptacles.find {
+                it.location.name == location && it.materialUuid != null
+            }
+            if (receptacle) {
+                receptacle.location.name
+            }
+        }.findAll()
+
+        if (occupiedLocations.size() > 0) {
+            throw new TransferException("The following locations already occupied in the destination labware: ${occupiedLocations.join(', ')}")
```

Comment:
You can move the getting the location name part of the collect statement here to avoid finding the sames if they're unneeded.